src/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {useCallback, useRef, useState} from 'react'
+import {List, GroupedListProps, ListPropsBase, ItemInput} from '../ActionList/List'
+import Overlay from '../Overlay'
+import {DropdownButton, DropdownButtonProps} from './DropdownButton'
+import {Item} from '../ActionList/Item'
+import {useFocusTrap} from '../hooks/useFocusTrap'
+import {useFocusZone} from '../hooks/useFocusZone'
+import {useAnchoredPosition} from '../hooks/useAnchoredPosition'
+import {useRenderForcingRef} from '../hooks/useRenderForcingRef'
+import randomId from '../utils/randomId'
+
+export interface DropdownMenuProps extends Partial<Omit<GroupedListProps, keyof ListPropsBase>>, ListPropsBase {
+  /**
+   * A custom fuction component used to render the anchor element.
+   * Will receive the selected text as `children` prop when an item is activated.
+   * Uses a `DropdownButton` by default.
+   */
+  renderAnchor?: <T extends React.HTMLAttributes<HTMLElement>>(props: T) => JSX.Element
+
+  /**
+   * A placeholder value to display on the trigger button when no selection has been made.
+   */
+  placeholder?: string
+
+  /**
+   * An `ItemProps` item from the list of `items` which is currently selected.  This item will receive a checkmark next to it in the menu.
+   */
+  selectedItem?: ItemInput
+
+  /**
+   * A callback which receives the selected item or `undefined` when an item is activated in the menu.  If the activated item is the same as the current
+   * `selectedItem`, `undefined` will be passed.
+   */
+  setSelectedItem?: (item?: ItemInput) => unknown
```

Comment:
Could we call this `onChange` to be more consistent with other components in React like `<input>` and `<select>`?

```suggestion
  onChange?: (item?: ItemInput) => unknown
```

Reply:
Changed in https://github.com/primer/components/pull/1135/commits/7e74ab11283dd3f7de2668a5aebb777190ef0df4